### PR TITLE
Webapp update

### PR DESCRIPTION
**Pull request by 16BitNarwhal**
following discussion of #2502 
- exposed two new routes, `/indexes` and `/indexes/{index}/status` from the server
- webapp dropdown list is generated from `/indexes` route
- dropdowns use index corpus + method instead of indexname field now
- dropdowns have indicator for whether they're already cached
- `/index/{index}/` is renamed to `/indexes/{index}`

image

### CODE REVIEW

File: src/main/java/io/anserini/server/Controller.java [hunk 1]
```diff
@@ -52,4 +59,34 @@ public Map<String, Object> search(@PathVariable(value = "index", required = fals
     return queryMap;
   }
 
+  @RequestMapping(method = RequestMethod.GET, path = "/isCached/{index}")
```
Comment (lintool) [inline src/main/java/io/anserini/server/Controller.java:62]: See https://medium.com/@nadinCodeHat/rest-api-naming-conventions-and-best-practices-1c4e781eb6a5

What's the best practice for a "status endpoint"?

File: src/main/java/io/anserini/server/Controller.java [hunk 2]
```diff
@@ -52,4 +59,34 @@ public Map<String, Object> search(@PathVariable(value = "index", required = fals
...
+    if (!IndexInfo.contains(index)) {
+      throw new IllegalArgumentException("Index name " + index + " not found!");
+    }
+    
+    PrebuiltIndexHandler handler = new PrebuiltIndexHandler(index);
+    handler.initialize();
+    return handler.checkIndexFileExist();
+  }
+
+  @RequestMapping(method = RequestMethod.GET, path = "/list")
```
Comment (lintool) [inline src/main/java/io/anserini/server/Controller.java:73]: Same comment... what are some alternative route names that might better conform to best practices?

**Comment by lintool** [comment]
Also, can you attach here a screenshot so I have an idea of what it looks like now?

**Comment by 16BitNarwhal** [comment]
> Additional thoughts about API design: maybe the route should be `catalog`? and asking whether or not it's downloaded should be a param?

Whoops, saw this a bit late. I was thinking `/index/{index}/cached` to check status and `/indexes` to list all

**Comment by lintool** [comment]
I think this is a good time to document the API in an MD, e.g., `docs/rest-api.md` or something like that?

Provide examples like this? https://github.com/castorini/anserini/blob/master/docs/fatjar-regressions/fatjar-regressions-v0.36.1.md#webapp-and-rest-api

**Review by lintool — APPROVED** [review]

**Pull request merged**